src/object.rs

```diff
@@ -36,18 +37,18 @@ impl PyObject {
             !ptr.is_null() && ffi::Py_REFCNT(ptr) > 0,
             format!("REFCNT: {:?} - {:?}", ptr, ffi::Py_REFCNT(ptr))
         );
-        PyObject(ptr)
+        PyObject(NonNull::new_unchecked(ptr))
     }
 
     /// Creates a `PyObject` instance for the given FFI pointer.
     /// Panics if the pointer is `null`.
     /// Undefined behavior if the pointer is invalid.
     #[inline]
-    pub unsafe fn from_owned_ptr_or_panic(py: Python, ptr: *mut ffi::PyObject) -> PyObject {
+    pub unsafe fn from_owned_ptr_or_panic(_py: Python, ptr: *mut ffi::PyObject) -> PyObject {
```

Comment:
Removing the unused `_py` arg leads to `pyfunction` macro breaking in a way I'm not familiar enough to address now.

Reply:
It's because you need to have the GIL acquired or the python c api calls done by panic_after_error and the debug version of from_owned_ptr are undefined behavior.